drivers/flash/spi_flash_mx25.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2016 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief This file defines the private data structures for spi flash driver
+ */
+
+#ifndef __SPI_FLASH_MX25_H__
+#define __SPI_FLASH_MX25_H__
+
+
+struct spi_flash_data {
+	struct device *spi;
+	struct spi_config spi_cfg;
+	struct k_sem sem;
+};
+
```

Comment:
Since there is so few structs and all defined, you could merge the content of spi_flash_mx25_defs.h here.